mathesar/models/base.py

```diff
@@ -283,6 +283,57 @@ class Explorations(BaseModel):
     description = models.CharField(null=True)
 
 
+class Form(BaseModel):
+    id = models.UUIDField(primary_key=True)
+    name = models.CharField()
+    version = models.IntegerField()
+    database = models.ForeignKey('Database', on_delete=models.CASCADE)
+    base_table_oid = models.PositiveBigIntegerField()
+    schema_oid = models.PositiveBigIntegerField()
+    description = models.CharField(null=True)
+    header = models.OneToOneField('FormHeader', on_delete=models.DO_NOTHING)
+    submission = models.ForeignKey('SubmissionSettings', on_delete=models.CASCADE)
+    is_public = models.BooleanField(default=False)
+
+
+class FormHeader(BaseModel):
+    title = models.CharField()
+    subtitle = models.CharField(null=True)
+
+
+class SubmissionSettings(BaseModel):
+    role = models.ForeignKey('ConfiguredRole', on_delete=models.CASCADE)
+    message = models.CharField(null=True)
+    redirect_url = models.URLField(null=True)
+    submit_label = models.CharField(null=True)
+
+
+class FormField(BaseModel):
+    id = models.CharField()
+    attnum = models.SmallIntegerField()
+    related_form = models.ForeignKey('Form', on_delete=models.CASCADE)
```

Comment:
This is not an entire sub-Form, it's only a list of FormFields.

Reply:
The way I understand this is that each form can have multiple form fields. I added a foreign key on `Form` model, So, anytime we list forms we can filter the related FormFields by form ids. Am I missing something?

Reply:
Yes, you're right with how the FK should work. I mistook this as something representing `nested_fields`, which seems to be missing.